source/opt/strip_atomic_counter_memory_pass.h

```diff
@@ -22,11 +22,26 @@
 namespace spvtools {
 namespace opt {
 
-// DNS: Write documentation
+// Removes the AtomicCounterMemory bit from the value being passed into memory
+// semantics. This bit being set is ignored in Vulkan environments and
+// forbidden WebGPU ones.
 class StripAtomicCounterMemoryPass : public Pass {
  public:
   const char* name() const override { return "strip-atomic-counter-memory"; }
   Status Process() override;
+
```

Comment:
Can someone with a deeper understanding of the optimizer confirm this list?
I removed kAnalysisDefUse and  kAnalysisConstants, since I am pretty sure this pass may change things related to those analyses.

Reply:
The constant manager will still be valid.  The def-use manager requires a single line to keep it up-to-date.  I would suggest that you do that.  It can make a noticeable difference in compile time if you need to rebuild it a lot.